lib/stack_master/commands/apply.rb

```diff
@@ -12,7 +12,7 @@ def initialize(config, stack_definition, options = Commander::Command::Options.n
         @stack_definition = stack_definition
         @from_time = Time.now
         @options = options
-        @options.on_failure ||= "ROLLBACK"
+        @options.on_failure ||= @config.stack_defaults['on_failure']
```

Comment:
This means if you set a stack default for `on_failure` SAM templates won't work.

Since this never worked, can we just set it to `nil` by default? I'll add some documentation in a seperate PR that this doesn't work and that's expected